### PR TITLE
id support for args

### PR DESCRIPTION
**Pull request by zeromomentum121**
## About

adds ID parsing support for args
i don't know what i'm doing
One thing I think should be added as well but I thought was too breaking for some people's code was changing mentionX to just X (eg: mentionUser -> user)

## Status

- [ ] These changes have been tested against Discord API or contain no API change.
- [ ] This PR includes only documentation changes, no code change.
- [possible] This PR introduces some Breaking changes.

### CODE REVIEW

**Comment by MierenManz** [comment]
Oh I see what you did. Now these don't only match `<!@5728157285727582>` but also `5728157285727582`
That is actually really smart. It might be handy to write a test for it tho. Check `./test/argsparser_test.ts` for that
This change may need to wait for #149 because that changed a bit of code aswell.

I really like the idea tho and it is definitely a good suggestion

**Comment by MierenManz** [comment]
Idk about changing the match type tho. They are both fine to me but removing the "mention" with this pr might be a good one because the parsing for it is now not only limited to mentions but also id's

**Review by Helloyunho — APPROVED** [review]
LGTM! This is really a good idea

**Comment by DjDeveloperr** [comment]
> They are both fine to me but removing the "mention" with this pr might be a good one because the parsing for it is now not only limited to mentions but also id's

This change would make sense

**Comment by zeromomentum121** [comment]
> > They are both fine to me but removing the "mention" with this pr might be a good one because the parsing for it is now not only limited to mentions but also id's
> 
> This change would make sense

✔️ commited that

**Comment by MierenManz** [comment]
Okay. Then this can probably be merged once #149 landed

**Comment by MierenManz** [comment]
@zeromomentum121 could you maybe add a test aswell? Just copy one of the tests in `./test/argsparser_test.ts` and remove the `<@!>` part from it. Just to make sure that it works correctly and that we can find bugs if we change stuff

**Comment by DjDeveloperr** [comment]
Please resolve conflicts and are tests updated with new match type names?

**Review by minseo0388 — CHANGES_REQUESTED** [review]
it's a good idea but now conflicts with current things

File: test/argsparser_test.ts [hunk 1]
```diff
@@ -108,15 +108,15 @@ Deno.test({
...
   },
   {
     name: 'channel',
-    match: 'mentionChannel'
+    match: 'channel'
   },
   {
     name: 'role',
-    match: 'mentionRole'
+    match: 'channel'
```
Comment (MierenManz) [inline test/argsparser_test.ts:116]: Mention role to channel?

**Review by MierenManz — CHANGES_REQUESTED** [review]

**Review by zeromomentum121 — COMMENTED** [review]
fixed

File: test/argsparser_test.ts (hunk 1 above)
Comment (zeromomentum121) [reply test/argsparser_test.ts]: fixed, i think i messed up when using find and replace

**Review by minseo0388 — APPROVED** [review]

**Pull request merged**